modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/internal/display/context/SelectAssetDisplayPageDisplayContext.java

```diff
@@ -297,9 +300,14 @@ public boolean inheritableDisplayPageTemplate() {
 			return _inheritableDisplayPageTemplate;
 		}
 
+		Bundle bundle = FrameworkUtil.getBundle(getClass());
+
+		BundleContext bundleContext = bundle.getBundleContext();
+
 		LayoutDisplayPageProviderRegistry layoutDisplayPageProviderRegistry =
-			LayoutDisplayPageProviderRegistryUtil.
-				getLayoutDisplayPageProviderRegistry();
+			bundleContext.getService(
```

Comment:
@slnn When you get service you should always unget it.

Reply:
@slnn Also, please avoid using this way. Move the static field into the display context class, and use a component to inject the referenced service.